From 48a0758c9deb0e8db598ff4d90eef1c54b36adf1 Mon Sep 17 00:00:00 2001
From: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Date: Thu, 1 Mar 2018 10:36:51 +0200
Subject: [PATCH 3416/5242] MLK-17626: drm: imx: dcss: fix "ctxld error"
 messages

commit  5b97e53e35ede9c60fe80a3420db1174a58c6fcb from
https://source.codeaurora.org/external/imx/linux-imx.git

The problem arised because of a combination of 2 commits:

Commit 1:

"2a70f32 - MLK-17232-2: drm: imx: dcss: ignore SB_PEND_DISP_ACTIVE
interrupt"

disabled the SB_PEND_DISP_ACTIVE interrupt because of a problem in SOC.
However, it did not remove the flag from CTXLD_IRQ_ERROR macro.

Commit 2:

"f0e3911 - MLK-17459-1: drm: imx: dcss: change ctxld irq handling"

moved the bottom half interrupt handling to top half. By doing that, the
top half did not exit immediately if IRQ_COMPLETION condition was met
and continued evaluating if any interrupts in CTXLD_IRQ_ERROR flags
were triggered.

This patch removes SB_PEND_DISP_ACTIVE interrupt flag from
CTXLD_IRQ_ERROR macro.

Signed-off-by: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/dcss/dcss-ctxld.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/gpu/imx/dcss/dcss-ctxld.c b/drivers/gpu/imx/dcss/dcss-ctxld.c
index 9ff1416..c6c4eb3 100644
--- a/drivers/gpu/imx/dcss/dcss-ctxld.c
+++ b/drivers/gpu/imx/dcss/dcss-ctxld.c
@@ -56,8 +56,7 @@
 
 #define CTXLD_IRQ_NAME			"ctx_ld" /* irq steer irq name */
 #define CTXLD_IRQ_COMPLETION		(DB_COMP | SB_HP_COMP | SB_LP_COMP)
-#define CTXLD_IRQ_ERROR			(RD_ERR | DB_PEND_SB_REC | \
-					 SB_PEND_DISP_ACTIVE | AHB_ERR)
+#define CTXLD_IRQ_ERROR			(RD_ERR | DB_PEND_SB_REC | AHB_ERR)
 
 /* The following sizes are in entries, 8 bytes each */
 #define CTXLD_DB_CTX_ENTRIES		1024	/* max 65536 */
-- 
1.7.9.5

